ecr_api.py

```diff
@@ -173,6 +185,45 @@ def __init__ ( self , retries=60) :
             break
```

Comment:
The coverage results are showing this line as untested.  I don't know how that could be... how else are you escaping the `while True`?  oh well.